### PR TITLE
Fixed: #5896 Custom fields of type "CUSTOM REGEX" are always saved as "ANY"

### PR DESCRIPTION
**Pull request by jwhulette**
Fixed: Custom fields of type "CUSTOM REGEX" are always saved as "ANY"

### CODE REVIEW

File: app/Helpers/Helper.php [hunk 1]
```diff
@@ -225,8 +225,9 @@ public static function customFieldsetList()
      */
     public static function predefined_formats()
     {
-        $keys = array_keys(CustomField::$PredefinedFormats);
+        $keys = array_keys(CustomField::PREDFINED_FORMATS);
```
Comment (uberbrady) [inline app/Helpers/Helper.php]: typo.

File: app/Http/Controllers/CustomFieldsController.php [hunk 2]
```diff
@@ -158,7 +160,16 @@ public function edit($id)
 
         $this->authorize('update', $field);
 
-        return view("custom_fields.fields.edit")->with('field', $field);
+        // Check to see if there is a custom regex format type
+        $field->format_type = $field->format;
+        if(stripos($field->format,'regex') === 0 && ($field->format !== CustomField::PREDFINED_FORMATS['MAC'])) {
```
Comment (uberbrady) [inline app/Http/Controllers/CustomFieldsController.php]: oh, I guess the typo is in the constant name? Probably should fix.

File: app/Models/CustomField.php [hunk 3]
```diff
@@ -15,27 +15,32 @@ class CustomField extends Model
     use ValidatingTrait,
         UniqueUndeletedTrait;
 
+    /**
+     * Custom field predfined formats
+     * 
+     * @var array
+     */
+    const PREDFINED_FORMATS = [
```
Comment (uberbrady) [inline app/Models/CustomField.php]: Ah ha! Here it is.

**Comment by jwhulette** [comment]
I fixed the spelling  but something is still not working correctly. DO NOT MERGE YET.

**Comment by snipe** [comment]
The threading between code comments and regular comments here is confusing - is this ready for merging, or should I still hold off?

**Comment by jwhulette** [comment]
Please hold off on merging... I found a logic error after pushing the spelling changes. I hope to have it resolved in the next day or two.

**Comment by jwhulette** [comment]
I fixed the logic error. The PR is ready to be merged

**Comment by snipe** [comment]
Awesome, thanks!

**Pull request merged**